File: Resoluções OBI/OBI2020/python/piloto_automatico.py
# ler a entrada
a = int(input())
b = int(input())
c = int(input())

# variáveis que representam as diferenças entre as posições dos carros
diff1 = b-a
diff2 = c-b

if diff1 < diff2: # se o carro B está mais perto do A do que do C
    print(1)
elif diff1 > diff2: # se o carro B está mais perto do C do que do A
    print(-1)
else: # se o carro B está no exato meio entre os outros dois carros
    print(0)

File: Resoluções OBI/OBI2020/cpp/piloto_automatico.cpp
#include <bits/stdc++.h>
using namespace std;

// declarar variáveis da entrada
int a, b, c;

int main() {

    // ler a entrada
    cin >> a >> b >> c;
    
    // variáveis que representam as diferenças entre as posições dos carros
    int diff1 = b-a, diff2 = c-b;

    if(diff1 < diff2) { // se o carro B está mais perto do A do que do C
        cout << 1 << endl;
    } else if(diff1 > diff2) { // se o carro B está mais perto do C do que do A
        cout << -1 << endl;
    } else { // se o carro B está no exato meio entre os outros dois carros
        cout << 0 << endl;
    }

}
